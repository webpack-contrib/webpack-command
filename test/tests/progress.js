const webpack = require('webpack');

const progress = require('../../lib/progress');
const { test } = require('../util');

test('progress util', module, () => {
  // e.g. building modules|5/6 modules|1 active|/node_modules/lodash/lodash.js
  //      building modules|14/14 modules|0 active|
  //      finish module graph
  //      finish module graph|FlagDependencyExportsPlugin

  it('should parse args: all args', () => {
    const args = [
      false,
      '0.1',
      'building modules',
      '5/6 modules',
      '1 active',
      '/node_modules/lodash/lodash.js',
    ];
    const result = progress.parseArgs(...args);

    expect(result).toMatchSnapshot();
  });

  it('should parse args: no filename', () => {
    const args = [false, '0.1', 'building modules', '5/6 modules', '1 active'];
    const result = progress.parseArgs(...args);

    expect(result).toMatchSnapshot();
  });

  it('should parse args: only step name', () => {
    const args = [false, '0.1', 'finish module graph'];
    const result = progress.parseArgs(...args);

    expect(result).toMatchSnapshot();
  });

  it('should parse args: step name and scope', () => {
    const args = [
      false,
      '0.1',
      'finish module graph',
      'FlagDependencyExportsPlugin',
    ];
    const result = progress.parseArgs(...args);

    expect(result).toMatchSnapshot();
  });

  it('should apply', () => {
    const config = { profile: false };
    const compiler = webpack({});
    const reporter = { progress: () => {} };
    progress.apply(config, compiler, reporter);

    expect(compiler.hooks.compilation.taps).toMatchSnapshot();
  });
});
