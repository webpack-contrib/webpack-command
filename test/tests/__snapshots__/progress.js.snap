// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`progress util > should apply #0 1`] = `
Array [
  Object {
    "fn": [Function],
    "name": "FunctionModulePlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "NodeSourcePlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "LoaderTargetPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "JavascriptModulesPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "JsonModulesPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "WebAssemblyModulesPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "SingleEntryPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "CompatibilityPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "HarmonyModulesPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "AMDPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "CommonJsPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "LoaderPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "LoaderPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "NodeStuffPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "RequireJsStuffPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "APIPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "ConstPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "UseStrictPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "RequireIncludePlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "RequireEnsurePlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "RequireContextPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "ImportPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "SystemPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "EnsureChunkConditionsPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "RemoveParentModulesPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "RemoveEmptyChunksPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "MergeDuplicateChunksPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "FlagIncludedChunksPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "OccurrenceOrderPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "SideEffectsFlagPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "FlagDependencyExportsPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "FlagDependencyUsagePlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "ModuleConcatenationPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "NoEmitOnErrorsPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "DefinePlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "UglifyJSPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "TemplatedPathPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "RecordIdsPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "WarnCaseSensitiveModulesPlugin",
    "type": "sync",
  },
  Object {
    "fn": [Function],
    "name": "ProgressPlugin",
    "type": "sync",
  },
]
`;

exports[`progress util > should parse args: all args #0 1`] = `
Object {
  "fileName": "/node_modules/lodash/lodash.js",
  "profile": false,
  "scope": undefined,
  "step": Object {
    "index": 1,
    "modulePosition": 5,
    "name": "building modules",
    "percentage": 0.1,
    "state": "active",
    "totalModules": 6,
  },
}
`;

exports[`progress util > should parse args: no filename #0 1`] = `
Object {
  "fileName": undefined,
  "profile": false,
  "scope": undefined,
  "step": Object {
    "index": 1,
    "modulePosition": 5,
    "name": "building modules",
    "percentage": 0.1,
    "state": "active",
    "totalModules": 6,
  },
}
`;

exports[`progress util > should parse args: only step name #0 1`] = `
Object {
  "fileName": undefined,
  "profile": false,
  "scope": undefined,
  "step": Object {
    "index": undefined,
    "modulePosition": undefined,
    "name": "finish module graph",
    "percentage": 0.1,
    "state": undefined,
    "totalModules": undefined,
  },
}
`;

exports[`progress util > should parse args: step name and scope #0 1`] = `
Object {
  "fileName": undefined,
  "profile": false,
  "scope": "FlagDependencyExportsPlugin",
  "step": Object {
    "index": undefined,
    "modulePosition": undefined,
    "name": "finish module graph",
    "percentage": 0.1,
    "state": undefined,
    "totalModules": undefined,
  },
}
`;
